### PR TITLE
JBIDE-21810 - update openshift username/pw, respect values in .cdk file

### PR DESCRIPTION
**Pull request by robstryker**

### CODE REVIEW

**Comment by fbricon** [comment]
Loogs good but can you provide unit tests for that?

File: plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/listeners/CDKOpenshiftUtility.java [hunk 1]
```diff
@@ -46,9 +48,26 @@ public IConnection findExistingOpenshiftConnection(IServer server, ADBInfo adb)
 	}
 	
 	public IConnection createOpenshiftConnection(IServer server, ADBInfo adb) {
+		return createOpenshiftConnection(server, adb, true);
+	}
+	public IConnection createOpenshiftConnection(IServer server, ADBInfo adb, boolean add) {
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/listeners/CDKOpenshiftUtility.java]: better replace boolean by a ConnectionsRegistry parameter

File: plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/listeners/CDKOpenshiftUtility.java [hunk 2]
```diff
@@ -46,9 +48,26 @@ public IConnection findExistingOpenshiftConnection(IServer server, ADBInfo adb)
 	}
 	
 	public IConnection createOpenshiftConnection(IServer server, ADBInfo adb) {
+		return createOpenshiftConnection(server, adb, true);
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/listeners/CDKOpenshiftUtility.java]: return createOpenshiftConnection(server, adb, ConnectionsRegistrySingleton.getInstance());

File: plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/listeners/CDKOpenshiftUtility.java [hunk 3]
```diff
@@ -57,7 +76,13 @@ public IConnection createOpenshiftConnection(IServer server, ADBInfo adb) {
...
 		((Connection)con).setAuthScheme(authScheme);
 		((Connection)con).setUsername(username);
-		ConnectionsRegistrySingleton.getInstance().add(con);
+
+		if( password != null ) {
+			((Connection)con).setPassword(password);
+		}
+		
+		if( add ) 
+			ConnectionsRegistrySingleton.getInstance().add(con);
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/listeners/CDKOpenshiftUtility.java]: if (connectionRegistry != null) {
   connectionRegistry.add(con);
}

File: tests/org.jboss.tools.openshift.cdk.server.test/src/org/jboss/tools/openshift/cdk/server/test/internal/CDKOpenshiftUtilityTest.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+import junit.framework.TestCase;
+
+public class CDKOpenshiftUtilityTest extends TestCase {
+	@Test
+	public void testOpenshiftConnectionCredentials() throws Exception {
+		CDKOpenshiftUtility util = new CDKOpenshiftUtility();
+		ADBInfo adb = createADB();
+		IServer s = mockServer("openshift33");
+
```
Comment (fbricon) [inline tests/org.jboss.tools.openshift.cdk.server.test/src/org/jboss/tools/openshift/cdk/server/test/internal/CDKOpenshiftUtilityTest.java]: ConnectionRegistry registry = mock(ConnectionRegistry.class)

File: tests/org.jboss.tools.openshift.cdk.server.test/src/org/jboss/tools/openshift/cdk/server/test/internal/CDKOpenshiftUtilityTest.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+public class CDKOpenshiftUtilityTest extends TestCase {
+	@Test
+	public void testOpenshiftConnectionCredentials() throws Exception {
+		CDKOpenshiftUtility util = new CDKOpenshiftUtility();
+		ADBInfo adb = createADB();
+		IServer s = mockServer("openshift33");
+		
+		createCDKFile("Basic", null, null);
+		IConnection con = util.createOpenshiftConnection(s, adb, false);
```
Comment (fbricon) [inline tests/org.jboss.tools.openshift.cdk.server.test/src/org/jboss/tools/openshift/cdk/server/test/internal/CDKOpenshiftUtilityTest.java]: util.createOpenshiftConnection(s, adb, registry);

File: tests/org.jboss.tools.openshift.cdk.server.test/src/org/jboss/tools/openshift/cdk/server/test/internal/CDKOpenshiftUtilityTest.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+		assertNotNull(con);
+		assertEquals(con.getUsername(), "openshift-dev");
+		assertEquals(con.getPassword(), "devel");
```
Comment (fbricon) [inline tests/org.jboss.tools.openshift.cdk.server.test/src/org/jboss/tools/openshift/cdk/server/test/internal/CDKOpenshiftUtilityTest.java]: verify(registry).add(con);

File: tests/org.jboss.tools.openshift.cdk.server.test/src/org/jboss/tools/openshift/cdk/server/test/internal/CDKOpenshiftUtilityTest.java [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+		
+		createCDKFile("Basic", "test", null);
+		con = util.createOpenshiftConnection(s, adb, false);
```
Comment (fbricon) [inline tests/org.jboss.tools.openshift.cdk.server.test/src/org/jboss/tools/openshift/cdk/server/test/internal/CDKOpenshiftUtilityTest.java]: util.createOpenshiftConnection(s, adb, null);

File: tests/org.jboss.tools.openshift.cdk.server.test/src/org/jboss/tools/openshift/cdk/server/test/internal/CDKOpenshiftUtilityTest.java [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+		assertNotNull(con);
+		assertEquals(con.getUsername(), "test");
+		assertEquals(con.getPassword(), null);
+
+		createCDKFile("Basic", "test", "pass");
+		con = util.createOpenshiftConnection(s, adb, false);
```
Comment (fbricon) [inline tests/org.jboss.tools.openshift.cdk.server.test/src/org/jboss/tools/openshift/cdk/server/test/internal/CDKOpenshiftUtilityTest.java]: util.createOpenshiftConnection(s, adb, null);

File: tests/org.jboss.tools.openshift.cdk.server.test/src/org/jboss/tools/openshift/cdk/server/test/internal/CDKOpenshiftUtilityTest.java [hunk 9]
```diff
@@ -0,0 +1,125 @@
+package org.jboss.tools.openshift.cdk.server.test.internal;
```
Comment (fbricon) [inline tests/org.jboss.tools.openshift.cdk.server.test/src/org/jboss/tools/openshift/cdk/server/test/internal/CDKOpenshiftUtilityTest.java]: missing copyright header

**Comment by fbricon** [comment]
+1 to apply

**Comment by robstryker** [comment]
pushed to merge, then realized forgot to squash immediately,  then squashed and push-forced :D  don't kill me.

**Pull request merged**